server/lib/UserManager.php

```diff
@@ -270,7 +241,85 @@ public function register(Request $request, Response $response) {
 		return $response;
 	}
 
-	/**
+    public function delete(Request $request, Response $response) {
+        $body = json_decode($request->getBody(), true);
+
+        if ($body === null || !isset($body['message']) || !isset($body['message']['data']) ||
+            !isset($body['message']['data']['federationId']) || !isset($body['signature']) ||
+            !isset($body['message']['timestamp'])) {
+            $response->withStatus(400);
+            return $response;
+        }
+
+        $cloudId = $body['message']['data']['federationId'];
+
+        try {
+            $verified = $this->verifyRequest($cloudId, $body['message'], $body['signature']);
+        }  catch(\Exception $e) {
+            $response->withStatus(400);
+            return $response;
+        }
+
+
+        if ($verified) {
+            $result = $this->deleteDBRecord($cloudId);
+            if ($result === false) {
+                $response->withStatus(403);
+            }
+        } else {
+            // ERROR OUT
+            $response->withStatus(403);
+        }
+
+        return $response;
+    }
+
+    /**
+     * check signature of incoming request
+     *
+     * @param $cloudId
+     * @param $message
+     * @param $signature
+     * @return bool|Response
+     * @throws \Exception
+     */
+    protected function verifyRequest($cloudId, $message, $signature) {
+        // Get fed id
+        list($user, $host) = $this->splitCloudId($cloudId);
+
+        // Retrieve public key && store
+        $ocsreq = new \GuzzleHttp\Psr7\Request(
+            'GET',
+            'https://'.$host . '/ocs/v2.php/identityproof/key/' . $user,
+            [
+                'OCS-APIREQUEST' => 'true',
+                'Accept' => 'application/json',
+            ]);
+
+        $client = new Client();
+        $ocsresponse = $client->send($ocsreq, ['timeout' => 10]);
+
+        $ocsresponse = json_decode($ocsresponse->getBody(), true);
+
+        if ($ocsresponse === null || !isset($ocsresponse['ocs']) ||
+            !isset($ocsresponse['ocs']['data']) || !isset($ocsresponse['ocs']['data']['public'])) {
+            throw new \Exception();
```

Comment:
I think [BadMethodCallException](http://php.net/manual/de/class.badmethodcallexception.php) sounds good.
Anyway I was just forwarding the complain from phpstorm.